tests/unit/charms/lte_core_interface/v0/test_lte_core_interface_provider.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import unittest
+from ipaddress import AddressValueError
+
+import pytest
+from ops import testing
+
+from tests.unit.charms.lte_core_interface.v0.dummy_provider_charm.src.charm import (
+    DummyCoreProviderCharm,
+)
+
+testing.SIMULATE_CAN_CONNECT = True
+
+BASE_CHARM_DIR = "tests.unit.charms.lte_core_interface.v0.dummy_requirer_charm.src.charm.DummyCoreRequirerCharm"  # noqa: E501
+
+
+class TestCoreProvider(unittest.TestCase):
+    def setUp(self):
+        self.relation_name = "lte-core"
+        self.remote_app = "lte-core-requirer"
+        self.harness = testing.Harness(DummyCoreProviderCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_given_unit_is_leader_and_remote_unit_joined_relation_when_set_mme_address_then_data_is_added_to_application_databag(  # noqa: E501
+        self,
+    ):
+        self.harness.set_leader(is_leader=True)
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=self.remote_app
+        )
+        mme_ipv4_address = "0.0.0.0"
+
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=self.remote_app,
+            key_values={"mme_ipv4_address": mme_ipv4_address},
+        )
+
+        relation_data = self.harness.get_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.app.name
+        )
+        self.assertEqual(relation_data["mme_ipv4_address"], mme_ipv4_address)
+
+    def test_given_unit_is_not_leader_and_remote_unit_joined_relation_when_set_mme_address_then_data_is_not_added_to_application_databag(  # noqa: E501
+        self,
+    ):
+        self.harness.set_leader(is_leader=False)
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=self.remote_app
+        )
+        mme_ipv4_address = "0.0.0.0"
+
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=self.remote_app,
+            key_values={"mme_ipv4_address": mme_ipv4_address},
+        )
+
+        relation_data = self.harness.get_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.app.name
+        )
+        self.assertEqual(relation_data, {})
+
+    def test_given_relation_is_not_created_when_set_core_information_then_runtime_error_is_raised(  # noqa: E501
```

Comment:
Do we need to test this situation if `set_core_information` is run on `relation_joined`?

Reply:
Probably we don't. I thought something like an edge case in which the relation is created and broken super quickly. As I couldn't make sure something like that can happen, I tested it just in case. Do you see the test problematic or just useless?

Reply:
I'd rather say useless, because I don't see this situation happening.

Reply:
Dropped.